### DISCUSSION TITLE
4.0.0 launched!

### DISCUSSION THREAD

**Discussion by 0xfe** [0xfe/vexflow#1226] [General]
Hi folks, I just pushed 4.0.0-beta from Ron's branch! Thank you all for your hard work, particularly @ronyeh @rvilarl @AaronDavidNewman @tommadams @mscuthbert @h-sug1no who contributed immensely to this release!

I want to do a series of announcements: starting with a few tweets and a mini blog post going over what's new, what's changed, and a more detailed post about the migration from JS to TS, and some of our learnings.

Meanwhile, can you please send me your twitter handles, so I can tag you in the first tweet?

Also, send me other links you might want me to share in the blogpost (e.g., your website, blog, or anything else you want.)

**Comment by tommadams** [comment]
Woohoo! Congratulations to everyone for all their hard work. I'm really excited to get all the new features out into an official release.

I try and stay off social media and don't have a twitter account, but many thanks for the consideration :)

**Comment by ronyeh** [comment]
Awesome. I think your first blog post should mostly concentrate on the TypeScript support (and very soon, native ESM support).

All the other things are improvements / refinements but can be discussed in future blog posts or in our documentation. VexFlow 4 is mostly about modernizing the project, and I'm super happy that we're now in TS :-).

**Comment by mscuthbert** [comment]
Congrats on huge work!  My twitter is @mscuthbert also!
